### PR TITLE
CONSOLE 3132: Introduce new console-plugin-shared npm package

### PR DESCRIPTION
**Pull request by mareklibra**
The package will contain shared code to develop dynamic plugins which
can be linked at build-time and so does not need to be part of the
dynamic plugin API.

### CODE REVIEW

File: frontend/packages/console-plugin-shared/src/components/OverviewDetailItem.tsx
Comment (mareklibra) [inline frontend/packages/console-plugin-shared/src/components/OverviewDetailItem.tsx:14]: Thinking about a way to enforce that in case of `error !== undefined`. Any idea?

What about changing the `error` to just hold the message, so `error?: string`?

Comment (rawagner) [reply frontend/packages/console-plugin-shared/src/components/OverviewDetailItem.tsx:14]: I agree. `error?: string` is better.

File: frontend/packages/console-plugin-shared/OWNERS
Comment (rawagner) [inline frontend/packages/console-plugin-shared/OWNERS:6]: I think reviewers/approvers should be the same as https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/OWNERS and later we can expand

File: frontend/packages/console-plugin-shared/OWNERS
Comment (rawagner) [inline frontend/packages/console-plugin-shared/OWNERS:8]:
```suggestion
  - component/plugin-shared
```

File: frontend/packages/console-plugin-shared/package.json
Comment (rawagner) [inline frontend/packages/console-plugin-shared/package.json:8]:
```suggestion
  "homepage": "https://github.com/openshift/console/tree/master/frontend/packages/console-plugin-shared",
```

File: frontend/packages/console-plugin-shared/package.json
Comment (rawagner) [inline frontend/packages/console-plugin-shared/package.json:5]: we will need a script which will generate a publish-able npm package. We do the same for sdk https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/package.json#L11

File: frontend/packages/console-plugin-shared/src/components/OverviewDetailItem.css
Comment (rawagner) [inline frontend/packages/console-plugin-shared/src/components/OverviewDetailItem.css:10]: we should rename to
```suggestion
.co-overview-details-card__item-title {
  flex-basis: 100%;
}

.co-overview-details-card__item-value {
  flex-basis: 100%;
  margin-bottom: 8px;
  overflow: hidden;
  overflow-wrap: break-word;
}
```

**Review by rawagner — CHANGES_REQUESTED** [review]

File: frontend/packages/console-plugin-shared/src/components/OverviewDetailItem.tsx
Comment (mareklibra) [reply frontend/packages/console-plugin-shared/src/components/OverviewDetailItem.tsx:14]: done

File: frontend/packages/console-plugin-shared/OWNERS
Comment (mareklibra) [reply frontend/packages/console-plugin-shared/OWNERS:8]: done

File: frontend/packages/console-plugin-shared/package.json
Comment (mareklibra) [reply frontend/packages/console-plugin-shared/package.json:8]: oops, done

File: frontend/packages/console-plugin-shared/src/components/OverviewDetailItem.css
Comment (mareklibra) [reply frontend/packages/console-plugin-shared/src/components/OverviewDetailItem.css:10]: done

File: frontend/packages/console-plugin-shared/package.json
Comment (mareklibra) [reply frontend/packages/console-plugin-shared/package.json]: Let's use `private: false` and avoid generating package.json

File: frontend/webpack.config.ts
Comment (mareklibra) [inline frontend/webpack.config.ts:139]: TODO: this does not work as expected, tests are failing

File: frontend/packages/console-plugin-shared/package.json
Comment (rawagner) [inline frontend/packages/console-plugin-shared/package.json:2]: we will be publishing under openshift-console npm org
```suggestion
  "name": "@openshift-console/plugin-shared",
```

Comment (mareklibra) [reply frontend/packages/console-plugin-shared/package.json]: done

**Comment by rawagner** [comment]
/lgtm
/hold

@spadgett PTAL.

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by rawagner** [comment]
/label px-approved
/label docs-approved

there are no user visible changes. The PR adds a new package which will be published to npmjs and plugin developers will be able to use components and utility functions from this package

**Comment by yapei** [comment]
did a regression testing about cluster details, node details and project details which utilizes the new `OverviewDetailItem` component from console-plugin-shared npm package, no regressions found
/label qe-approved

**Pull request merged**
